test/e2e/certificates_test.go

```diff
@@ -0,0 +1,102 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/certificates"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// testCertificates tests the CA certificates for Windows nodes
+// The initial kubelet CA certificate is valid for 1 year, from the date of the cluster installation.
+// Usually, the first rotation of the kubelet CA certificate is generated by the [API Server Operator](https://github.com/openshift/cluster-kube-apiserver-operator)
+// at 80% for the CA certificate lifecycle (1 year), which is 292 days, and removes the previous CA certificate after 365 days.
+func testCertificates(t *testing.T) {
+	// init test context
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	t.Run("Kubelet CA rotation", tc.testKubeletCARotation)
+}
+
+// testKubeletCARotation tests the rotation of the kubelet CA certificate by forcing the generation of a new CA
+// certificate and ensuring the content of the CA bundle in all Windows nodes contains the new certificate. Once the
+// rotation is triggered, WMCO detects the new CA certificate, merge it with the initial bundle, and copy the resulting
+// bundle to all Windows nodes.
+//
+// This test accounts for the kubelet CA automatic rotation that takes place at 80% of the initial certificate's
+// lifecycle (1 year).  Note that the old certificate is still present in the cluster until day 365, so clients may be
+// able to use it.
+func (tc *testContext) testKubeletCARotation(t *testing.T) {
+	// force CA rotation
+	err := tc.forceKubeApiServerCertificateRotation()
+	require.NoError(t, err)
+
+	// wait 5 seconds for the new certificate and add 5 seconds per node to allow
+	// WMCO to update the kubelet-ca.crt file on each node
+	nodeCount := len(gc.allNodes())
+	time.Sleep(time.Duration(5+5*nodeCount) * time.Second)
```

Comment:
This is going to be prone to flaking, I'd prefer to see a retry loop when checking each of the Nodes

as in a retry within the sub test `t.Run("node/"+winNode.Name, func(t *testing.T) {`

Reply:
Addressed.